projects/igniteui-angular/src/lib/data-operations/pivot-strategy.ts

```diff
@@ -151,7 +151,7 @@ export class DimensionValuesFilteringStrategy extends FilteringStrategy {
         const dim = enabledDimensions.find(x => x.memberName === column.field);
         const allValuesHierarchy = PivotUtil.getFieldsHierarchy(
             data,
-            [dim],
+            dim ? [dim] : [],
```

Comment:
This will do in this case, but maybe if we want to fix it further we could use the possibility to improve the `flatten` function that it is used deeper in the chain to not perform any actions on elements that are undefined as well?